src/test/fuzz/util.h

```diff
@@ -512,4 +512,63 @@ void ReadFromStream(FuzzedDataProvider& fuzzed_data_provider, Stream& stream) no
     }
 }
 
+class FuzzedSocket : public MockableSocket
+{
+    FuzzedDataProvider& m_fuzzed_data_provider;
+
+public:
+    explicit FuzzedSocket(FuzzedDataProvider& fuzzed_data_provider) : m_fuzzed_data_provider{fuzzed_data_provider}
+    {
+    }
+
+    ssize_t send(const char* buf, size_t len, int flags) override
+    {
+        if (m_fuzzed_data_provider.ConsumeBool()) {
+            return len;
+        }
+        return m_fuzzed_data_provider.ConsumeIntegralInRange<ssize_t>(-1, len);
+    }
+
+    ssize_t recv(char* buf, size_t len, int flags) override
+    {
+        if (buf == nullptr || len == 0 || m_fuzzed_data_provider.ConsumeBool()) {
+            return m_fuzzed_data_provider.ConsumeBool() ? 0 : -1;
+        }
+        const std::vector<uint8_t> random_bytes = m_fuzzed_data_provider.ConsumeBytes<uint8_t>(len);
+        if (random_bytes.empty()) {
+            return m_fuzzed_data_provider.ConsumeBool() ? 0 : -1;
```

Comment:
Here and above, set `errno` to some "random" value if we are going to return `-1`. Maybe the "random" value should have higher probablitity of being one of `EAGAIN`, `EWOULDBLOCK`, `EINTR` or `EINPROGRESS`.

Reply:
Good point! Pushed an updated version addressing this.